torch/lib/libshm/core.cpp

```diff
@@ -79,17 +87,20 @@ ClientSocket& get_manager_socket(char *manager_handle) {
 
 char * copy_handle(const std::string &handle) {
   char *new_handle = new char[handle.length()+1];
-  std::strcpy(new_handle, handle.c_str());
+  memcpy(new_handle, handle.c_str(), handle.length() + 1);
   return new_handle;
 }
 
 AllocInfo get_alloc_info(libshm_context *ctx) {
   AllocInfo info = {0};
   info.pid = getpid();
   info.free = false;
-  // TODO: make sure full name has fit into the buffer
-  char *filename = THMapAllocatorContext_filename(ctx->th_context);
-  strncpy(info.filename, filename, sizeof(info.filename));
+  const char *filename = THMapAllocatorContext_filename(ctx->th_context);
+  size_t len = strlen(filename);
+  if (len >= sizeof(info.filename)) {
+    throw std::runtime_error("THMapAllocatorContext_filename too long");
+  }
+  memcpy(info.filename, filename, len + 1);
```

Comment:
FYI, the strncpy to memcpy changes are because of this:
http://meyering.net/crusade-to-eliminate-strncpy/

The FB linter gets upset at strncpy calls. Basically, it doesn't give you the guarantees you'd expect (mostly NULL terminating the string).